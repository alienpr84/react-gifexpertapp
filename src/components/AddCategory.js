import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleOnChangeInputValue = (e) => {
		setInputValue(e.target.value);
	}

	const handleOnSubmit = (e) => {
		e.preventDefault();
		
		if(inputValue.trim().length > 2) {
			setCategories( prev => [inputValue, ...prev]);
			setInputValue('');
		}
		
	}

	return (
		<div>
			<form onSubmit={handleOnSubmit}>
				<input 
					value= { inputValue }
					onChange={ handleOnChangeInputValue }
					placeholder="Enter a value"
				/>
			</form>
		</div>
	)
}

AddCategory.prototype = {
	setCategories: PropTypes.func.isRequired,
}