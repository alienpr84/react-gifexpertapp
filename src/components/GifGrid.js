import useFetchGifs from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs( category )

  return (
    <div>
      <h3>{category}</h3>

			{ loading && 'loading...' }

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem
						key={img.id}
						{...img}
					/>
        ))}
      </div>
    </div>
  );
};

export default GifGrid;
