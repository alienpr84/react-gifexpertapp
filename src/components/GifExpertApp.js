import { useState } from 'react';
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One punch']);

  return (
    <>
      <h2>GifExpertApp</h2>
			<AddCategory setCategories={ setCategories } />
      <hr />

			<ul>
			{
				categories.map( category => <GifGrid key={ category } category={ category }/>)
			}
			</ul>
    </>
  );
};

export default GifExpertApp;
