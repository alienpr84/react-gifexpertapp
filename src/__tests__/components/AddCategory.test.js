import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Testing on <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper;

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});

  test("Must show <AddCategory /> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change the text on the box", () => {
    const input = wrapper.find("input");
    const value = "Hunter";

    input.simulate("change", { target: { value } });
  });

	test('cant post the information with submit', () => {
		wrapper.find('form').simulate('submit', { preventDefault(){} });

		expect( setCategories ).not.toHaveBeenCalled();
	});

	test('must call the setCategories and clean the text box', () => {
		const value = 'Boruto'

		wrapper.find('input').simulate('change', { target: { value }});
		wrapper.find('form').simulate('submit', { preventDefault(){} });
		
		expect( setCategories ).toHaveBeenCalled();
		
		const inputValue = wrapper.find('input').prop('value');

		expect( inputValue ).toBe('');
	});
});
