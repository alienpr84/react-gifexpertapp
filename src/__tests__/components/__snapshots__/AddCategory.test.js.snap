// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <AddCategory /> Must show <AddCategory /> correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Enter a value"
      value=""
    />
  </form>
</div>
`;
